Tetfuza/TetfuzaCLI/SystemConsoleDisplay.cs

```diff
@@ -0,0 +1,128 @@
+            // Loop through array and have each line centered on the screen
+            for (int i = 0; i < boardView.Length - 1; i++)
+            {
+                int location = i;
+                //if (_cheatCode.ToLower() == "up") // A small Easter Egg to make the pieces fall up
+                //    location = boardView.Length - 2 - i;
+                WriteText("| " + boardView[location] + "|");
+            }
+            WriteText("-----------------------");
+            WriteText(" ");
+            WriteText("Z and X to rotate");
+            WriteText("Arrow keys to move (left, right, down)");
+        }
+
+        public void WriteText(string text)
+        {
+            int width = Console.WindowWidth;
+            int textBeginIndex = (width / 2) - (text.Length / 2) - 1;
+            for (int space = 0; space < textBeginIndex; space++)
+            {
+                Console.Write(" ");
+                text += " ";
+            }
+            Console.WriteLine(text);
+        }
+
+        /// <summary>
+        /// Allow user to choose to start on any level between 0 and 19
+        /// </summary>
+        /// <returns></returns>
+        public int DrawMenu(string prompt)
+        {
+            Console.CursorVisible = false;
+            int startLevel = -1;
+            bool validInput = false;
+            //_cheatCode = Console.ReadLine().ToLower();
+            ClearScreen();
+            do
+            {
+                WriteText(prompt);
+                string input = Console.ReadLine();
+                validInput = (int.TryParse(input, out startLevel) && startLevel >= 0 && startLevel <= 19);
+            } while (!validInput);
+            //if (_cheatCode == "impossible")
+            //{
+            //    startLevel = 29;
+            //}
+            return startLevel;
+        }
+
+        public void ClearScreen()
+        {
+            Console.Clear();
+        }
+
+        public void RedrawFrame()
+        {
+            Console.SetCursorPosition(0, 0);
+        }
+
+        /// <summary>
+        /// Displays the upcoming piece
+        /// </summary>
+        private void DrawNextPiece(FuzaPiece nextPiece)
+        {
+            
+            string[] pieceView = nextPiece.ToString().Replace(FuzaPiece.FUZA_CHAR, TetfuzaBoard.MOVING_CHAR).Split(")");
+            WriteText("Next Piece:");
+            for (int i = 0; i < pieceView.Length - 1; i++)
+            {
+                int location = i;
+                //if (_cheatCode.ToLower() == "up") // A small Easter Egg to make the pieces appear upsidedown
+                //    location = pieceView.Length - 2 - i;
+                WriteText(pieceView[location]);
+            }
+        }
+        /// <summary>
+        /// Converts a large number to a string with normal comma separation
+        /// </summary>
+        /// <param name="num"></param>
+        /// <returns></returns>
+        public string CommasInNumber(long num)
```

Comment:
This needs tested. I noticed some weird comma splicing in playing the game yesterday.

Reply:
Plus I think there’s probably a library function that formats a number with commas

Reply:
There is a test written for this, and it is confirmed working as intended. I think the weird comma separation you saw had to do with the character ghosting that happens when you hold down a key. The key input into the console causes the rest of that line to be drawn 1 character offset to the right, and because the screen isn't cleared between frames, if that 1 character to the right isn't overwritten then it will linger indefinitely. The original CenterText() method had this issue, and the fix to this accidentally introduced the newline issue... one problem always leads to another

Reply:
That’s probably right